mule-extensions-archetype/src/main/resources/archetype-resources/src/main/java/internal/__extensionName__ConnectionProvider.java

```diff
@@ -1,4 +1,4 @@
-package ${package};
+package ${package}.internal;
```

Comment:
the internal shouldn't be necesary

Reply:
either that or we change the folder structure. this looks OK for me